articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+---
+uid: CreateNewDecorator
+---
+# How can I create a new Decorator?
+
+Click [here](xref:Decorator) to find out more about Decorators.
+
+This how-to will serve as an example of how to create a new Decorator that will inject code at runtime for a Template that has been setup for it.
+
+This how-to will build up from the [previous](xref:MakeTemplateDecoratorReady) how-to, which showed how to make a Template ready for Decorators. Please interpolate on your end what is written here to make the necessary adjustments on what you're trying to achieve.
+
+## Create a new Decorator
+
+Typically you would create a new Decorator in a new Intent Architect Application which will reference a Template from another Module or Intent Architect Application. As such we will be doing that.
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
+
+Once the generation is done, create a new Decorator called `StaticFileServerDecorator`.
+
+![New Decroator](images/create-new-decorator/NewDecorator.png)
+
+We need to specify the Contract name for that Decorator so that it will know for which Template it is meant to Decorate.
```

Comment:
```suggestion
Enter `MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract` in the `Implements Decorator Contract`'s `Type Fullname` field.
```

Side note, could we change the module builder stereotype to make this a drop down with templates in the module, would we want to?

Reply:
Unfortunately not, unless we make it a Type of its own where people can define a fully qualified namespace to the Contract type they are referring to. That might be the best but not sure how much will need to change.

Reply:
If I recall there are other places in this how-to which maybe highlight that there may be room for improvement in Intent or the Module Builder. Maybe there is nothing more we can do here, but any friction point in the process should be scrutinized for possible improvement and GitHub issues made where possible.

Reply:
Definitely, and let's keep track of it using the Issue logging feature on Github

Reply:
I've made an issue at https://github.com/IntentSoftware/IntentArchitect/issues/188.

Let me know how you feel about the suggested change or if you're happy feel free to commit it yourself.